addons/purchase_stock/models/stock_rule.py

```diff
@@ -44,6 +44,17 @@ def _onchange_action(self):
         if self.action == 'buy':
             self.location_src_id = False
 
+    def _get_po_line_grouping_key(self, po_line):
```

Comment:
Please use the same signature in your module: `def _get_po_line_grouping_key(self, line):`
`po_line` is the better choice

Reply:
You also need to use `@api.model` since this function does not depend on the recordset of `self`